agent/pool/pool.go

```diff
@@ -387,6 +389,26 @@ func (p *ConnPool) dial(
 	return conn, hc, nil
 }
 
+func (p *ConnPool) RPCClientTimeout() time.Duration {
+	// Since clientTimeout represents uint32 milliseconds,
+	// convert to int64 first then multiply by 1000 to
+	// match time.Duration's nanoseconds.
+	return time.Duration(int64(atomic.LoadUint32(p.clientTimeout) * 1000))
```

Comment:
Also, I'd suggest `time.Millisecond` instead of the magic 1000. So:

```go
return time.Duration(atomic.LoadUint32(&p.clientTimeout)) * time.Millisecond
```